velox/serializers/UnsafeRowSerializer.cpp

```diff
@@ -112,11 +112,31 @@ void UnsafeRowVectorSerde::deserialize(
     RowVectorPtr* result,
     const Options* /* options */) {
   std::vector<std::optional<std::string_view>> serializedRows;
+  std::vector<std::string> concatenatedRows;
+
   while (!source->atEnd()) {
     // First read row size in big endian order.
     auto rowSize =
         folly::Endian::big(source->read<UnsafeRowVectorSerializer::TRowSize>());
     auto row = source->nextView(rowSize);
+
+    // If we couldn't read the entire row at once, we need to concatenate it
+    // manually.
+    if (row.size() < rowSize) {
+      std::string rowBuffer;
+      rowBuffer.reserve(rowSize);
+      rowBuffer.append(row);
+
+      while (rowBuffer.size() < rowSize) {
+        auto rowFragment = source->nextView(rowSize - rowBuffer.size());
+        VELOX_CHECK_GT(
+            rowFragment.size(), 0, "Unable to read full serialized UnsafeRow.");
```

Comment:
perhaps, include the full size and how much we could read in the error message for easier troubleshooting